lib/src/gif/web_gif_exporter.dart

```diff
@@ -0,0 +1,60 @@
+import 'package:flutter/material.dart';
+import 'package:image/image.dart' as image;
+import 'dart:ui' as ui show ImageByteFormat;
+import 'package:flutter/foundation.dart';
+import 'package:screen_recorder/src/frame.dart';
+import 'package:screen_recorder/src/gif/gif_exporter.dart';
+
+GifExporter gifExporter() => WebGifExporter();
+
+class WebGifExporter implements GifExporter {
+  final List<Frame> _frames = [];
+
+  @override
+  Future<List<int>?> export() async {
+    if (_frames.isEmpty) {
+      return null;
+    }
+    List<RawFrame> bytes = [];
+    for (final frame in _frames) {
+      final i = await frame.image.toByteData(format: ui.ImageByteFormat.png);
```

Comment:
This part still will freeze the screen on Web if you have many frames @ueman

Reply:
Yeah I know. For now it's just an improvement for io platforms. For web it's just the old implementation. (This PR is a few commits behind)

Reply:
The problem is that this operation(toByteData) we dont have any way to include it on a Web Worker, thats why my other proposal of processing the frames on the way. I dont like much my approach but honestly i dont find any other workaround :S

Reply:
So then, for me to get context about the PR, where is the improvement exactly? Capturing or just exporting?

Reply:
This significantly improves the gif exporting on io platforms. Each frame is directly processed for a gif in a background isolate which means when the user actually requests the gif export, it's almost completely converted to a gif. So for a user this seems like a significant performance improvement. 

But as said, it just works on IO platforms and only for gifs as dart:ui APIs can't be used in a background isolate. 
But I'm guessing you're more interested in individual frames than a gif 😅

Reply:
Ok see what you mean now.

But yeah, this will still be a pain in my side haha Wondering what middle point we can have. For any reason i dont understand why the operation toByteData is freezing just on Web not in iPhone.